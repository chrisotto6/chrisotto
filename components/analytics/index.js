import GA from './GoogleAnalytics'
import siteMetadata from '@/data/siteMetadata'

const isProduction = process.env.NODE_ENV === 'production'

const Analytics = () => {
  return <>{isProduction && siteMetadata.analytics.googleAnalyticsId && <GA />}</>
}

export default Analytics
