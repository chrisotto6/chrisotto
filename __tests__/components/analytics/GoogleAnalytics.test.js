import { render } from '@testing-library/react'
import GAScript from '@/components/analytics/GoogleAnalytics'
import { logEvent } from '@/components/analytics/GoogleAnalytics'

it('renders GAScript unchanged', () => {
  const { container } = render(<GAScript />)
  expect(container).toMatchSnapshot()
})

it('logs the event', () => {
  const windowSpy = jest.spyOn(window, 'window', 'get')
  windowSpy.mockImplementation(() => ({
    location: {
      origin: 'https://example.com',
    },
  }))
  logEvent('action', 'category', 'label', 1)
  windowSpy.mockRestore()
})
