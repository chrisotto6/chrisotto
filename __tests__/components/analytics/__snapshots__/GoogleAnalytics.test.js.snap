// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GAScript unchanged 1`] = `<div />`;
